### PR TITLE
Add first draft of prototype README

### LINKED ISSUE #480: Prototype: More welcoming README!
We want to have a more polished Readme inviting users to interact with the Stackable Universe!

To figure out what a good readme looks like for all operators, we will spike it in the secret-operator and in the kafka-operator.

The point is to experiment, and see what content we want, to later make sure we can template a similar layout for all operators.

Inspiration of good READMEs which we could try to imitate: [ChrunchyData/postgres-operator](https://github.com/CrunchyData/postgres-operator), [datahub-project/datahub](https://github.com/datahub-project/datahub)

It could make sense to spike multiple README versions and get feedback.

## Ideas (not requirements)

- **connect to user intentions when they visit the readme**
    - find out what the tool is
    - no-nonsense information for technical users, what does it _do_
        - It reads a CR and creates XYZ
    - is the tool well maintained and reliable?
        - make it professional and a bit flashy (the company logo is a good signal)
- it should be visual, not a wall of text
    - maybe a gif of an install?
- maybe a diagram?
    - maybe this one? https://stackable.tech/wp-content/uploads/2022/06/illustration-40.jpg
- show that the operator is part of a larger platform and integrates well with the other operators (i.e. kafka needs a zookeeper and the SDP includes a zookeeper operator)
    - list all operators?
- this is not for you if ...
- Quickstart
    - stackablectl
- which flavors of k8s do we support (k3s, managed k8s, Openshift (soon))
- Contributing
- link to blog posts if it makes sense
- (companies that use the product)

## Required Content (by topic, not necessarily by title)

The new README version should contain information/sections about:

- Introduction
- Quickstart (stackablectl)
- Technical no-nonsense how does it work
- About the Stackable Data Platform
- List of other operators
- Contributing
- Further Reading

## Acceptance Criteria

- [x] Created at least one new file called README_v1.md with above content.
- [x] The actual content of the sections does not need to be polished or final. Think one step above lorem ipsum.
- [x] The new README can incorporate information of the existing README.md, as well as the new required sections in a nice flow
- [x] You MAY jot down considerations of how the new README could be recreated using templating techniques for future tasks

### PR DESCRIPTION
**Pull request by vsupalov**
# Description

A draft of what a new type of README could look like for this operator. Resolves #480

This PR is meant for discussion and future reference, not for eventual merging.

## Review Checklist

- [ ] Code contains useful comments
- [ ] CRD change approved (or not applicable)
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: README_v1.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+We love open-source!
+
+# Supported Platforms
+
+We develop and test our operators for the following cloud platforms:
+
+* Kubernetes 1.21-1.24
+* Amazon EKS
+* Google Kubernetes Engine (GKE) on GCP
+* Microsoft AKS
```
Comment (maltesander) [inline README_v1.md:86]: Would it make sense to add this via badge? Kubernetes support and on which managed kubernetes the integration tests work? Maybe rust built version etc. as well?

**Review by maltesander — COMMENTED** [review]
Yeah looks much better already. 
I would add the stackable logo on top similar to https://github.com/stackabletech/secret-operator/pull/190 as well as a diagram for dependencies / resources.

File: README_v1.md (hunk 1 above)
Comment (fhennig) [reply README_v1.md:86]: I thought about this too, I think it could be nice. But I would also keep it here anyways

Comment (vsupalov) [reply README_v1.md:86]: A badge could be cool. I'd worry that it looks too much like it's closely associated. But I'm not a lawyer.

**Comment by vsupalov** [comment]
> I would add the stackable logo on top similar to https://github.com/stackabletech/secret-operator/pull/190 as well as a diagram for dependencies / resources.

I liked them a lot as well! Then I tried to forget about them and start from scratch. The diagrams and logo are pretty dope (just as the "link bar" with all important links, so cool). I would include them in our upcoming template.

File: README_v1.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
+# Stackable Operator for Apache Kafka
+
+[![Build Actions Status](https://ci.stackable.tech/job/kafka%2doperator%2dit%2dnightly/badge/icon?subject=Integration%20Tests)](https://ci.stackable.tech/job/kafka%2doperator%2dit%2dnightly)
```
Comment (fhennig) [inline README_v1.md:3]: I really like the Jenkins status! Definitely want to keep that

File: README_v1.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+[![Maintenance](https://img.shields.io/badge/Maintained%3F-yes-green.svg)](https://GitHub.com/stackabletech/kafka-operator/graphs/commit-activity)
+
+This is a Kubernetes Operator to manage [Apache Kafka](https://kafka.apache.org/) clusters.
+
+Looking to dive in? It's easy to get started with the Stackable data platform using `stackablectl`:
+
+```
+stackablectl spin up a demo
```
Comment (fhennig) [inline README_v1.md:11]: A "quickstart" right inside the introduction can be cool

File: README_v1.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+```
+
+Read more in [link to fancy demo using stackablectl](#).
```
Comment (fhennig) [inline README_v1.md:14]: This is something we still need

File: README_v1.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+## Installation
+
+To get the best experience, we recommended way to get started with Stackable is through the `stackablectl` tool.
+
+Read [the instructions below](#) to get started.
```
Comment (fhennig) [inline README_v1.md:20]: Where would this link to? I think we can link to the docs from here. I think I'd like lots of links to the docs in general, as that _should_ be the main hub of information

File: README_v1.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+## Getting Started
+
+To create a Kafka cluster with three nodes, you can follow this [tutorial](https://docs.stackable.tech/kafka/stable/getting_started/first_steps.html).
+
+Otherwise, give it a try with the [stackablectl](https://docs.stackable.tech/home/stable/getting_started.html) CLI tool!
+
+## Documentation
```
Comment (fhennig) [inline README_v1.md:28]: dedicated docs section, nice

File: README_v1.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+The documentation for this operator can be found at <https://docs.stackable.tech/kafka/stable/index.html>.
+
+The documentation for all Stackable products can be found at <https://docs.stackable.tech>.
+
+## What Does This Do?
+
+Operator looks at custom resources in the cluster, and makes sure apps are started according to those custom resources.
```
Comment (fhennig) [inline README_v1.md:36]: This is a different angle than I took, a bit more general (There's a resource and the operator deploys stuff). I think both has its place, I'm not so sure how much previous knowledge we should assume. Maybe we could have something more surface-level here and link to detailed explanations in the docs?

File: README_v1.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+
+If those resources change, the operator makes sure that the right actions are taken so the app is the way you want it to be.
+
+Here's what a custom resource for a Kafka cluster can look like:
+
+```
+---
+apiVersion: kafka.stackable.tech/v1alpha1
+kind: KafkaCluster
```
Comment (fhennig) [inline README_v1.md:45]: Definitely would keep a code block like this with an actual yaml resource

File: README_v1.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+
+Want to get started? Read below.
+
+## Behind the Scenes
+
+It is written by [Stackable](https://www.stackable.tech) in Rust.
+
+It uses kube-rs to talk to Kubernetes.
+
+We test it [extensively](https://ci.stackable.tech/) using [Kuttl](https://kuttl.dev/)-powered integration tests on managed Kubernetes of multiple cloud platforms and our own bare-metal clusters.
```
Comment (fhennig) [inline README_v1.md:67]: Testing is good to mention

File: README_v1.md [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+
+## About the Stackable Data Platform
+
+Stackable makes it easy to operate data applicationsin any Kubernetes cluster.
+
+The data platform offers many operators, new ones being added continuously. All our operators are designed and built to be easily interconnected and to be consistent to work with.
+
+Stackable GmbH is the company behind the Stackable Data Platform. Offering professional services, paid support plans and development.
+
+We love open-source!
```
Comment (fhennig) [inline README_v1.md:77]: We love open source! A good point to make

**Review by fhennig — APPROVED** [review]
All the required content as per the ticket is in here, all the acceptance criteria are fullfilled

File: README_v1.md (hunk 5 above)
Comment (vsupalov) [reply README_v1.md:20]: I'd link to the section in the README right below the line. It's a tiny thing, but personally I appreciate being able to click the text I want to go to.

File: README_v1.md (hunk 7 above)
Comment (vsupalov) [reply README_v1.md:36]: Yeah, maybe it's too high-level. We could assume that somebody stumbling on this knows what an operator is.

File: README_v1.md (hunk 8 above)
Comment (vsupalov) [reply README_v1.md:45]: I really like showing a bit of the internal working for people who scroll through. I would keep the yaml as light as possible, to show that it's easy to use the operator.

**Comment by vsupalov** [comment]
Changes have landed in https://github.com/stackabletech/issues/issues/292. Closing.

**Pull request closed without merging**